src/modules/flow/iio/nodes.c

```diff
@@ -505,6 +516,22 @@ magnet_open(struct sol_flow_node *node, void *data, const struct sol_flow_node_o
     SOL_INT_CHECK_GOTO(ret, >= (int)sizeof(mdata->iio_base.config.sampling_frequency_name), err);
     SOL_INT_CHECK_GOTO(ret, < 0, err);
 
+    mdata->iio_base.config.oversampling_ratio_table = calloc(4, sizeof(struct sol_str_table));
+    SOL_NULL_CHECK(mdata->iio_base.config.oversampling_ratio_table, -ENOMEM);
+
+    table = mdata->iio_base.config.oversampling_ratio_table;
+    table->key = strdup("in_magn_x_");
```

Comment:
you have to check the return of `strdup`, it can lead to a segv because you doing in the next line a `strlen` with it.

Since you'll need to do this with the others elements, you can create a macro here to avoid repeat code.